mysql-test/suite/innodb_gis/t/rollback.test

```diff
@@ -475,3 +475,23 @@ update t1 set a=point(5,5), b=point(5,5), c=5 where i < 3;
 insert into t1 values(5, point(5,5), point(5,5), 5);
 
 drop table t1;
+
+--echo #
+--echo # MDEV-35420 Server aborts while deleting the record
+--echo #               in spatial index
+--echo #
+CREATE TABLE t1 (c POINT NOT NULL, SPATIAL(c)) engine=InnoDB;
+CHECK TABLE t1;
+INSERT INTO t1  VALUES(ST_GeomFromText('POINT(114368751 656950466)'));
+BEGIN;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+ROLLBACK;
```

Comment:
Could we speed up the test by executing the following?
```sql
SET STATEMENT unique_checks=0,foreign_key_checks=0 FOR
START TRANSACTION;
INSERT INTO t1 SELECT ST_GeomFromText('POINT(114368751 656950466)') FROM seq_1_to_512;
ROLLBACK;
```
The way the test is currently written, it will use row-level undo logging as well as acquire a large number of explicit locks. We should avoid `INSERT…SELECT` reads from a persistent InnoDB table to work around [MDEV-24813](https://jira.mariadb.org/browse/MDEV-24813).

Maybe we do need the row-level undo logging so that `ROLLBACK` will execute the interesting piece of code.